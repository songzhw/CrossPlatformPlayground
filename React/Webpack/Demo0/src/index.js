import _ from "lodash";
import "../css/index.css"

function component() {
  const element = document.createElement('div');
  element.setAttribute("id", "app");
  element.innerHTML = _.join(['Hello', 'webpack', "222"], ' ');
  return element;
}

document.body.appendChild(component());
