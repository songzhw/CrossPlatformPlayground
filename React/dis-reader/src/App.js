import React, { Component } from "react";
import "./index.css";
import { http } from "./utils/download/HttpEngine";
import * as JSZip from "jszip";
import { JSDOM } from "jsdom";

class App extends Component {
  state = { text: "unload" };


  render() {
    return (
      <div className="appRootContainer">
        <span>{this.state.text}</span>
        <button className="buttonInHome" onClick={this.onDownload}>download epub</button>
        <button className="buttonInHome" onClick={this.onParseXML}>parse xml</button>
        <button className="buttonInHome" onClick={this.onUnzipContainer}>unzip epub: META-INF/container.xml</button>
        <button className="buttonInHome" onClick={this.onUnzipToc}>unzip epub: get ToC</button>
        {/*TODO read xml*/}
        {/*TODO render xhtml*/}
      </div>
    );
  }

  onDownload = () => {
    const url = "https://gerhardsletten.github.io/react-reader/files/alice.epub";
    http(url)
      .then(arraybufferData => {
        this.arraybufferResponse = arraybufferData;
        this.setState({ text: "donwload finished" });
      });
  };

  onUnzipContainer = () => {
    const path = "META-INF/container.xml";
    JSZip.loadAsync(this.arraybufferResponse)
      .then(zip => {
        return zip.file(path)
          .async("string");
      })
      .then(text => {
        console.log("szw content = ", text);
        this.setState({ text });
      });
  };

  onParseXML = () => {
    let xml = `
      <?xml version="1.0" encoding="UTF-8"?>
      <container xmlns="urn:oasis:names:tc:opendocument:xmlns:container" version="1.0">
        <rootfiles>
          <rootfile full-path="OPS/package.opf" media-type="application/oebps-package+xml">szw</rootfile>
        </rootfiles>
        <rootfile> 200 </rootfile>
      </container>
      `;
    const dom = new JSDOM(xml);
    const node = dom.window.document.querySelector("rootfile");
    console.log(`szw node text = ${node.textContent}`); //=> szw
    console.log(`szw node path = ${node.getAttribute("full-path")}`); //=> OPS/package.opf
    this.setState({ text: node.textContent });

  };

  onUnzipToc = () => {

  };
}

export default App;
