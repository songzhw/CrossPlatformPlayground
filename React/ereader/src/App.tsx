import React, { Component } from "react";
import { EpubView } from "./epubView/EpubView";

class App extends Component {
  public render() {
    return (
      <div className="App">
        <EpubView
          url="https://gerhardsletten.github.io/react-reader/files/alice.epub"
        />
      </div>
    );
  }
}

export default App;
