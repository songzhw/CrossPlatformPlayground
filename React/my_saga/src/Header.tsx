import React from "react";
import { Link, NavLink } from "react-router-dom";
import "./index.css";

export const Header = () => {
  return (
    <header className="header">
      <nav>
        <NavLink to="/first" className="header-link" activeClassName="header-link-active">Fetches</NavLink>
      </nav>
    </header>
  );
};