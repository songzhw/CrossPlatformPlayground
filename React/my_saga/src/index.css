body {
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
  'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
  sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
  monospace;
}

.header {
  background-color: #2e7d32;
  height: 20px;
  padding-left: 10px;
  padding-bottom: 10px;
  padding-top: 10px;
  color: white;
}

.header-link {
  color: #fff;
  font-size: 18px;
  text-decoration: none;
}

.header-link-active {
  border-bottom: #ebebeb solid 2px;
}