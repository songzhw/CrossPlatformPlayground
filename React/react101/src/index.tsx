import React from "react";
import ReactDOM from "react-dom";
import "./index.css";
import { RouterWrap } from "./Rtutor/router/RoutesForFirst";
import { RouterForNetwork } from "./Rtutor/network_zip/RouterForNetwork";
import RouterChap6 from "./reacts_book/chap6/Routes";
import RouterChap7 from "./reacts_book/chap7/Routes";
import { FavoritePetScreen } from "./Rtutor/persist/useLocalStorage/FavoritePetScreen";
import { CryptoScreen } from "./Rtutor/crypto/CryptoScreen";

ReactDOM.render(<CryptoScreen/>, document.getElementById("root"));

