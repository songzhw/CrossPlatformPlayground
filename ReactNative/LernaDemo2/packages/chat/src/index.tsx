import { NativeModules } from 'react-native';

type ChatType = {
  multiply(a: number, b: number): Promise<number>;
};

const { Chat } = NativeModules;

export default Chat as ChatType;
