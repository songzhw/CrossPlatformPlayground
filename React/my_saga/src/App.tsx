import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { Header } from "./Header";
import { FirstScreen } from "./screen/first/FirstScreen";

const Routers = () => {
  return (
    <BrowserRouter>
      <div>
        <Header/>
        <Switch>
          <Route path="/first" component={FirstScreen}/>
        </Switch>
      </div>
    </BrowserRouter>
  );
};


const App: React.FC = () => {
  return (
    <div className="App">
      <Routers/>
    </div>
  );
};

export default App;
