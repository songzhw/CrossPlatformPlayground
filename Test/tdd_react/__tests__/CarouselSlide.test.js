import React from "react";
import { shallow } from "enzyme";
import { CarouselSlide } from "../src/CarouselSlide";

describe("CaraouselSlide", () => {
  test("renders a <figure>", () => {
    const wrapper = shallow(<CarouselSlide/>);
    expect(wrapper.type()).toBe("figure");
  });

  test("render a <image/> child", () => {
    const wrapper = shallow(<CarouselSlide/>);
    expect(wrapper.childAt(0).type()).toBe("img");
    expect(wrapper.childAt(1).type()).toBe("figcaption");
  });

  test("test props from image, text", () => {
    const imageUrl = "http://a.png";
    const wrapper = shallow(<CarouselSlide/>);
    wrapper.setProps({ imageUrl });
    wrapper.setProps({ description: "hello", author: "szw" });

    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(imageUrl);

    const fig = wrapper.find("figcaption strong");
    expect(fig.text()).toBe("hello");

    const figAuthor = wrapper.find("figcaption");
    expect(figAuthor.text()).toBe("szw");
  });
});