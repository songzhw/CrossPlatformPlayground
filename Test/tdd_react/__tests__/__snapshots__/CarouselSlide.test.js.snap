// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaraouselSlide renders correctly 1`] = `ShallowWrapper {}`;

exports[`Styled Image (SImage) mounted SImage renders correctly 1`] = `ReactWrapper {}`;
